processing/src/main/java/org/apache/druid/segment/join/MapJoinableFactory.java

```diff
@@ -80,4 +80,21 @@ public boolean isDirectlyJoinable(DataSource dataSource)
     }
     return maybeJoinable;
   }
+
+  @Override
+  public Optional<byte[]> computeJoinCacheKey(DataSource dataSource)
```

Comment:
Right now as I see that additional info comes from the data source itself. The joinable factory is deciding based on the `dataSource` whether it can build the joinable factory.  can you share any hypothetical examples wherein one factory decides to build Joinable but the other one does not on the basis of `JoinConditionAnalysis`?